lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -277,8 +276,20 @@ func (c *Client) start() {
 
 		// C.read_rdp_output blocks for the duration of the RDP connection and
 		// calls handle_bitmap repeatedly with the incoming bitmaps.
-		if errCode := C.read_rdp_output(c.rustClient); errCode != C.ErrCodeSuccess {
-			c.cfg.Log.Warningf("Failed reading RDP output frame: %v", errCode)
+		res := C.read_rdp_output(c.rustClient)
+		message := C.GoString(res.message)
+		C.free_c_string(res.message)
+		if message != "" {
+			if res.err_code == C.ErrCodeSuccess {
+				c.cfg.Log.Warnf("RDP server disconnected with reason: %v", message)
```

Comment:
Sometimes the message here will be something benign, like "The disconnection was initiated by an administrative tool on the server in another session.", whereas other times it can be an error, like "An internal error has occurred in the Terminal Services licensing component.". It seems like overkill to distinguish between [all the possibilities](https://github.com/gravitational/rdp-rs/blob/389eaf7bb8f207c1e4f98ed3f2766816480ee912/src/core/global.rs#L281-L317) here, so I just went with a warning.

The api for `read_rdp_output` is admittedly a bit weird, its explained in the docstring. Open to suggestions for how to make it better

https://github.com/gravitational/teleport/blob/b7355f8bec35f8662239c23b9bf389ef7b0ad192/lib/srv/desktop/rdp/rdpclient/src/lib.rs#L1134-L1150

Reply:
The disconnect code you talked about helps. A small comment here could help too, something saying that sometimes the message might be an error like the one you described above, hence we go with a warning.

Reply:
Scratch all that, I added a function to distinguish between errors and just messages in rdp-rs and updated the handling and logging in 2598c7f3626e74013dd2b5fed943853db8691bd3